apps/extension/src/background/approvals/service.ts

```diff
@@ -21,89 +24,131 @@ export class ApprovalsService {
     protected readonly connectedTabsStore: KVStore<TabStore[]>,
     protected readonly keyRingService: KeyRingService,
     protected readonly ledgerService: LedgerService
-  ) { }
+  ) {}
 
   // Deserialize transfer details and prompt user
   async approveTransfer(txMsg: string, type?: AccountType): Promise<void> {
     const txMsgBuffer = Buffer.from(fromBase64(txMsg));
-    const id = uuid();
-    await this.txStore.set(id, txMsg);
+    const msgId = uuid();
+    await this.txStore.set(msgId, txMsg);
 
     // Decode tx details and launch approval screen
     const txDetails = deserialize(txMsgBuffer, TransferMsgValue);
-    const { source, target, token, amount: amountBN } = txDetails;
+    const {
+      source,
+      target,
+      token: tokenAddress,
+      amount: amountBN,
+      tx: { publicKey = "" },
+    } = txDetails;
     const amount = new BigNumber(amountBN.toString());
-    const baseUrl = `${browser.runtime.getURL(
-      "approvals.html"
-    )}#/approve-transfer`;
+    const baseUrl = `${browser.runtime.getURL("approvals.html")}#/approve-tx/${
+      TxType.Transfer
+    }`;
 
     const url = paramsToUrl(baseUrl, {
-      id,
+      msgId,
       source,
       target,
-      token,
+      tokenAddress,
       amount: amount.toString(),
-      type: type as string,
+      accountType: type as string,
+      publicKey,
     });
 
     this._launchApprovalWindow(url);
   }
 
   // Deserialize bond details and prompt user
-  async approveBond(
-    txMsg: string,
-    type: AccountType,
-    publicKey?: string
-  ): Promise<void> {
+  async approveBond(txMsg: string, type: AccountType): Promise<void> {
     const txMsgBuffer = Buffer.from(fromBase64(txMsg));
-    const id = uuid();
-    await this.txStore.set(id, txMsg);
+    const msgId = uuid();
+    await this.txStore.set(msgId, txMsg);
 
     // Decode tx details and launch approval screen
     const txDetails = deserialize(txMsgBuffer, SubmitBondMsgValue);
 
-    const { source, nativeToken: token, amount: amountBN } = txDetails;
+    const {
+      source,
+      nativeToken: tokenAddress,
+      amount: amountBN,
+      tx: { publicKey = "" },
+    } = txDetails;
     const amount = new BigNumber(amountBN.toString());
-    const baseUrl = `${browser.runtime.getURL("approvals.html")}#/approve-bond`;
+    const baseUrl = `${browser.runtime.getURL("approvals.html")}#/approve-tx/${
+      TxType.Bond
+    }`;
 
     const url = paramsToUrl(baseUrl, {
-      id,
+      msgId,
       source,
-      token,
+      tokenAddress,
       amount: amount.toString(),
-      publicKey: publicKey || "",
-      type: type as string,
+      publicKey,
+      accountType: type as string,
```

Comment:
I think `type as string` can be changed to `type` and the behaviour will be the same, since `type` is a non-optional parameter.

Reply:
I removed these, thanks!